### ISSUE TITLE
Add a HTTP Method conditionlet to our rules engine

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#23209]
TL;DR - allow rules to fire based on the HTTP Method sent.  For example, fire on PUT and POST requests, but not on GET requests.

Our rules engine is super powerful, but there are places where it should work that it does not.  Take this example:

When a browser makes a cross site REST call, it does a preflight request.  This is so the browser can verify the fact that the REST endpoint should actually be available for use from other sites - it does this by checking the CORS headers that are returned by the endpoint.  When a browser makes a preflight request, it sends the HTTP method "OPTIONS" in order to check the headers.  It would be ideal in managing headers to allow a customer to write a rule that says, "when a browser sends an OPTIONS request, add an appropriate `Access-Control-Allow-Origin`  header.

**Comment by fmontes** [comment]
OPTIONS in the method and Stop Processing in the action.

<img width="1334" alt="image" src="https://user-images.githubusercontent.com/751424/215799124-b419f57b-f1ab-4330-9ea4-a9c1ee8ce966.png">

**Comment by bryanboza** [comment]
Fixed, tested on release-23.02 // Docker // FF
